prql-compiler/src/ast/item.rs

```diff
@@ -93,6 +89,21 @@ pub struct Range {
     pub end: Option<Box<Node>>,
 }
 
+impl Range {
+    pub const fn unbounded() -> Self {
+        Range {
+            start: None,
+            end: None,
+        }
+    }
+
+    pub fn new(start: Option<i64>, end: Option<i64>) -> Self {
+        let start = start.map(|x| Box::new(Node::from(Item::Literal(Literal::Integer(x)))));
```

Comment:
Re the `Integer` — I had done a PR to let ranges take any node; e.g. 

```elm
from employees
filter (dob | in @2019-01-01..@2020-01-01)
```

goes to 

```sql
SELECT
  employees.*
FROM
  employees
WHERE
  dob BETWEEN DATE '2019-01-01'
  AND DATE '2020-01-01'
```

If that doesn't work any longer then no great stress; I can modify in a follow up

Reply:
This is still possible. `fn new` is just a helper function for constructing integer ranges. Should be named `new_int`.